src/lib/cimbar_js/cimbar_js.cpp

```diff
@@ -159,19 +149,18 @@ int cimbare_configure(unsigned color_bits, unsigned ecc, int compression, bool l
 		return initRes;
 
 	// check if we need to refresh the stream
-	bool refresh = (color_bits != _colorBits or ecc != _ecc or compression != _compressionLevel or legacy_mode != _legacyMode);
+	bool refresh = (mode_val != _modeVal or compression != _compressionLevel);
```

Comment:
One interesting exception I'm leaving in: we can still override compression on the encoder side.

The decoder *doesn't* care how much compression is used -- just that the data is in a zstd compressed frame. So if you wanted to, you could for example crank the compression to the maximum level (22), take the performance hit on the sender side, and possibly save a few bytes over the air.

... in practice I'm not sure how much it really would help, though.